api/apiclient_test.go

```diff
@@ -62,6 +62,34 @@ func (s *apiclientSuite) TestDialAPIToModel(c *gc.C) {
 	assertConnAddrForModel(c, location, info.Addrs[0], s.State.ModelUUID())
 }
 
+func (s *apiclientSuite) TestDialAPIToModelParallel(c *gc.C) {
+	info := s.APIInfo(c)
+	info.Addrs = append(info.Addrs, "1.2.3.4:17070")
+
+	n := 10
+	errors := make(chan error, n)
+	for i := 0; i < n; i++ {
+		go func() {
+			conn, _, err := api.DialAPI(info, api.DialOpts{
+				DialTimeout: 100 * time.Millisecond,
+			})
+			if err == nil {
+				conn.Close()
+			}
+
+			select {
+			case errors <- err:
+			case <-time.After(100 * time.Millisecond):
+			}
```

Comment:
What's the point of the second case? `err` will always get pushed into `errors`.